### PR TITLE
Use boolean fields instead of HashMap lookups for enabled checks

### PR DESCRIPTION
**Pull request by not-my-profile**

### CODE REVIEW

**Comment by not-my-profile** [comment]
@charliermarsh could you benchmark this?

**Comment by charliermarsh** [comment]
It makes a tiny difference (compared to 300.2ms / 301.4ms before):

Screen Shot 2023-01-15 at 4 26 49 PM

Could still be worth it? Guessing it simplifies the hash implementation, and the client code reads a bit more cleanly IMO. What do you think?

**Comment by charliermarsh** [comment]
(When I said "10% improvement" in that other PR, I think I was wrong. I was setting `enabled` to `FxHashSet::default()`, but then forcing `use_ast` to `true` to ensure we still did the parse + AST walk. But, that probably has a bunch of other side effects that impact performance -- e.g., we wouldn't be surfacing _any_ checks, so check reporting would be a no-op, and so on. I'm guessing what you have here is the limit on gains from this codepath.)

File: src/settings/mod.rs [hunk 1]
```diff
@@ -253,7 +256,7 @@ impl Settings {
             allowed_confusables: FxHashSet::from_iter([]),
             builtins: vec![],
             dummy_variable_rgx: Regex::new("^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$").unwrap(),
-            enabled: FxHashSet::from_iter([rule_code.clone()]),
+            enabled: FxHashSet::from_iter([rule_code.clone()]).into(),
```
Comment (charliermarsh) [inline src/settings/mod.rs:259]: We should do the same thing for `fixable`, if we can.

**Comment by not-my-profile** [comment]
I think it would be interesting to measure how much time is actually spend in the ruff codebase ... I would assume that much more time is spend in the RustPython parser. How much of a difference does it make when we don't invoke `check_ast` at all?

**Comment by charliermarsh** [comment]
I haven’t benchmarked it recently but the parser is the majority of the time. Invoking just the parser is at least 200ms, maybe more like 250ms (out of the 300ms above).

**Comment by not-my-profile** [comment]
Closing in favor of 816c5f9d37c7f41cd9042e0180843b3206b8c49d from #1930.

**Pull request closed without merging**